src/rgw/rgw_op.cc

```diff
@@ -6112,8 +6112,9 @@ void RGWCompleteMultipart::execute(optional_yield y)
   if (op_ret < 0)
     return;
 
-  // remove the upload obj
-  int r = meta_obj->delete_object(this, s->obj_ctx, y);
+  // remove the upload meta object ; the meta object is not versioned
+  // when the bucket is, as that would add an unneeded delete marker
+  int r = meta_obj->delete_object(this, s->obj_ctx, y, true /* prevent versioning */);
```

Comment:
I guess this is ok;  i have sense that I haven't really thought through all the implications of our multipart logic on zipper apis in general